pycalphad/core/eqsolver.pyx

```diff
@@ -82,6 +81,50 @@ cdef bint add_new_phases(object composition_sets, object removed_compsets, objec
         return True
     return False
 
+@cython.boundscheck(False)
+cdef int argmax(double* a, int a_shape) nogil:
+    cdef int i
+    cdef int result = 0
+    cdef double highest = -1e30
+    for i in range(a_shape):
+        if a[i] > highest:
+            highest = a[i]
+            result = i
+    return result
+
+def add_nearly_stable(object composition_sets, dict phase_records,
```

Comment:
Does it make any difference for this to be `cdef` or `cpdef`?

Reply:
I don't have an explanation for this, but I see a 15-20% increase (regression) in the run-time of the test suite when I change this function to `cpdef`. The difference is noisy but repeatable.